src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -56,5 +57,44 @@ class DefaultSource extends RelationProvider {
 
     CsvRelation(path, headerFlag, delimiterChar, quoteChar)(sqlContext)
   }
+
+  /**
+   * Creates a new relation for data store in CSV given parameters and user supported schema.
+   * Parameters have to include 'path' and optionally 'delimiter', 'quote', and 'header'
+   */
+  def createRelation(
```

Comment:
Seems we can consolidate the common logics in this one the above `createRelation`?